macros/dbt_metrics/gen_metric_cte.sql

```diff
@@ -0,0 +1,84 @@
+
+{%- macro athena__gen_metric_cte(metric_name, grain, dimensions, secondary_calculations, start_date, end_date, relevant_periods, calendar_dimensions, treat_null_values_as_zero) %}
+
+, {{metric_name}}__final as (
+
+    {%- if not treat_null_values_as_zero -%}
+        {%- set metric_val = metric_name -%}
+    {%- else -%}
+        {%- set metric_val = "coalesce(" ~ metric_name ~ ", 0) as " ~ metric_name -%}
+    {%- endif %}
+    
+    select
+        {% if grain != 'all_time' %}
+        date_{{grain}},
+            {%- if secondary_calculations | length > 0 -%}
+                {% for period in relevant_periods %}
+        parent_metric_cte.date_{{ period }},
+                {%- endfor -%}
+            {%- endif -%}
+        {%- endif -%}
+        
+        {%- for calendar_dim in calendar_dimensions %}
+        parent_metric_cte.{{ calendar_dim }},
+        {%- endfor %}
+
+        {%- for dim in dimensions %}
+        parent_metric_cte.{{ dim }},
```

Comment:
I think this also needs to to be chopped to just `{{ dim }}`

Reply:
Nice catch! Fixed it.

Any fields that is in the `using` clause should be chopped. I also did it for `parent_metric_cte.{{ calendar_dim }}` to just `{{ calendar_dim }}`